Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
+        }
+
+        /// <summary>
+        ///     Allows server to fully load in another scene on top of current active scene.
+        /// </summary>
+        /// <param name="scenePath">The full path to the scene file or the name of the scene.</param>
+        /// <param name="players">List of player's that are receiving the new scene load.</param>
+        /// <param name="shouldClientLoadNormally">Should the clients load this additively too or load it full normal scene change.</param>
+        public void ServerLoadSceneAdditively(string scenePath, INetworkPlayer[] players, bool shouldClientLoadNormally = false)
+        {
+            if (!Server && !Server.Active) return;
+
+            ServerSceneLoading(scenePath, players, shouldClientLoadNormally, SceneOperation.LoadAdditive);
         }
 
         /// <summary>
-        /// Called from ChangeServerScene immediately before NetworkSceneManager's LoadSceneAsync is executed
-        /// <para>This allows server to do work / cleanup / prep before the scene changes.</para>
+        ///     Allows server to fully unload a scene additively.
         /// </summary>
-        /// <param name="scenePath">Path of the scene that's about to be loaded</param>
-        internal void OnServerChangeScene(string scenePath, SceneOperation operation)
+        /// <param name="scene">The scene handle which we want to unload additively.</param>
+        /// <param name="players">List of player's that are receiving the new scene unload.</param>
+        /// <param name="shouldClientUnloadNormally">Should the clients unload this additively too or unload it full normal scene change.</param>
+        public void ServerUnloadSceneAdditively(Scene scene, INetworkPlayer[] players, bool shouldClientUnloadNormally = false)
         {
-            logger.Log("OnServerChangeScene");
+            if (!Server && !Server.Active) return;
+
+            ServerSceneUnLoading(scene, players, SceneOperation.UnloadAdditive);
+        }
+
+        /// <summary>
+        ///     When player authenticates to server we send a message to them to load up main scene and
+        ///     any other scenes that are loaded on server.
+        ///
+        ///     <para>Default implementation takes main activate scene as main and any other loaded scenes and sends it to player's
+        ///     Please override this function if this is not intended behavior for you.</para>
+        /// </summary>
+        /// <param name="player">The current player that finished authenticating.</param>
+        public virtual void OnServerAuthenticated(INetworkPlayer player)
+        {
+            logger.Log("[NetworkSceneManager] - OnServerAuthenticated");
+
+            string[] scenesPaths = new string[SceneManager.sceneCount - 1];
+
+            for (int sceneIndex = 0; sceneIndex < SceneManager.sceneCount; sceneIndex++)
+            {
+                if (SceneManager.GetActiveScene().buildIndex == sceneIndex) continue;
```

Comment:
it is not obvious why this is using build index

Reply:
All scenes have a build index. If the user is authenticating it should be fine to use build index at first start because you should be loading up first initial scene. Do you want a comment to say so in method?.